util/gpool/spmc/spmspool.go

```diff
@@ -0,0 +1,391 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spmc
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/resourcemanager/pooltask"
+	"github.com/pingcap/tidb/util/gpool"
+	"go.uber.org/zap"
+)
+
+// Pool is a single producer, multiple consumer goroutine pool.
+// T is the type of the task. We can treat it as input.
+// U is the type of the result. We can treat it as output.
+// C is the type of the const parameter. if Our task look like y = ax + b, C acts like b as const parameter.
```

Comment:
You mean here T， U and C may have relation like u=at + C ？ here is questions: 1) will this kind of things happen? 2) where is the input for formular like y= ax + b to connect the input, output and const parameter?

Reply:
Your consumer function will be in existence. they are created when the pool init. so you can save the const parameter into a consumer. but it is for all tasks. If you want to put const parameter into specific task. you can use C as this const parameter.